server/src/test/java/org/opensearch/common/settings/SettingTests.java

```diff
@@ -322,6 +324,48 @@ public void testValidator() {
         assertTrue(FooBarValidator.invokedWithDependencies);
     }
 
+    public void testRegexValidator() throws Exception {
+        // A regex that matches one or more digits
+        String expectedRegex = "\\d+";
+        Pattern expectedPattern = Pattern.compile(expectedRegex);
+        RegexValidator regexValidator = new RegexValidator(expectedRegex);
+
+        // Test that the pattern is correctly initialized
+        assertNotNull(expectedPattern);
+        assertNotNull(regexValidator.getPattern());
+        assertEquals(expectedPattern.pattern(), regexValidator.getPattern().pattern());
+
+        // Test that validate() throws an exception for invalid input
+        final RegexValidator finalValidator = new RegexValidator(expectedRegex);
+        assertThrows(IllegalArgumentException.class, () -> finalValidator.validate("\\d+"));
+
+        try {
+            regexValidator.validate("\\d+");
+        } catch (IllegalArgumentException e) {
+            fail("Expected validate() to not throw an exception, but it threw " + e);
```

Comment:
This test is [failing Gradle Check](https://build.ci.opensearch.org/job/gradle-check/11913/testReport/junit/org.opensearch.common.settings/SettingTests/testRegexValidator/):
> java.lang.AssertionError: Expected validate() to not throw an exception, but it threw java.lang.IllegalArgumentException: Setting must be valid [\d+],Custom setting matches regex [\d+]

You're trying to send in a string (which happens to be a regex but isn't treated as one there) as the setting value and it's failing (as expected).

Using a regular expression as the string to check is very confusing to other maintainers and does not correspond with the common use case.

The only place you should use the regex is constructing the validator, which you do with `new RegexValidator(expectedRegex)`.  Combined with the check `if (!pattern.matcher(value).matches())` that means you will fail if the regex does *not* match the input.  

The `\d+` regex matches 1 or more digits.  (In Java we need the double slashes to escape it for the string `"\\d+".)

So you should pass for something that is 1 or more digits like "123".  Just use "123" on line 343.  And use some other text string like "foo" or "text" on line 340 above.